addons/web_editor/static/src/js/wysiwyg/widgets/media_dialog.js

```diff
@@ -160,6 +161,13 @@ var MediaDialog = Dialog.extend({
         var args = arguments;
         return this.activeWidget.save().then(function (data) {
             self._clearWidgets();
+            // Restore classes if the media was replaced (when changing type)
+            if (self.media !== data) {
+                var oldClasses = self.media && _.toArray(self.media.classList);
+                if (oldClasses) {
+                    data.className = _.union(_.toArray(data.classList), oldClasses).join(' ');
+                }
+            }
```

Comment:
Is this still useful? Seems like a leftover from your old approach, right?

Reply:
Yes, this is still useful, as indicated by the comment, this is where classes are copied when the media-type changes. If you look at the icon save method, there used to be something there:

```js
var finalClasses = _.uniq(this.nonIconClasses.concat([iconFont.base, iconFont.font]));
```

To copy over non icon classes but since we want to copy the classes in all type changes (and not just changes to an icon) it makes sense to do it after calling the save method from the active widget.